python-sdk/src/astro/databricks/api_utils.py

```diff
@@ -0,0 +1,135 @@
+        output_file_path,
+    )
+
+    # Since we had to use jinja2 safe strings, we want to inline replace \' with '
+    # This is not a security concern now that jinja2 is no longer running.
+    for line in fileinput.input(output_file_path, inplace=True):
+        print(line.replace("\\'", "'"), end="")
+    log.debug("Created file %s", output_file_path)
+    return output_file_path
+
+
+def load_file_to_dbfs(local_file_path: Path, file_name: str, api_client: ApiClient) -> Path:
+    # TODO we should allow arbitrary dbfs paths set by env vars/users
+    """Load a file into DBFS. Used to move a python file into DBFS, so we can run the jobs as pyspark jobs.
+    Currently saves the file to dbfs:/mnt/pyscripts/, but will eventually support more locations.
+
+    :param local_file_path: path of the file to upload
+    :param api_client: Databricks API client
+    :return: path to the file in DBFS
+
+    """
+    log.debug("loading file %s", str(local_file_path))
+    dbfs = DbfsApi(api_client=api_client)
+    file_path = DbfsPath("dbfs:/mnt/pyscripts/")
+    dbfs.mkdirs(file_path)
+    dbfs.delete(file_path.join(file_name), recursive=False)
+    dbfs.put_file(src_path=str(local_file_path), dbfs_path=file_path.join(file_name), overwrite=True)
+    return Path("dbfs:/mnt/pyscripts") / file_name
+
+
+def create_and_run_job(
+    api_client,
+    file_to_run: str,
+    databricks_job_name: str,
+    existing_cluster_id=None,
+    new_cluster_specs=None,
+) -> None:
+    """Creates a databricks job and runs it to completion.
+
+    :param databricks_job_name: Name of the job to submit
+    :param file_to_run: path to file we will run with the job
+    :param api_client: databricks API client
+    :param existing_cluster_id: If you want to run this job on an existing cluster, you can give the cluster ID
+    :param new_cluster_specs: If you want to run this job on a new cluster, you can give the cluster specs
+        as a python dictionary.
+
+    """
+    jobs_api = JobsApi(api_client=api_client)
+    json_info = {
+        "name": databricks_job_name,
+        "spark_python_task": {"python_file": file_to_run},
+    }
+    if existing_cluster_id:
+        log.debug("Using existing cluster %s", existing_cluster_id)
+        json_info["existing_cluster_id"] = existing_cluster_id
+    elif new_cluster_specs:
+        log.debug("creating new spark cluster with spec", new_cluster_specs)
+        json_info["new_cluster"] = new_cluster_specs
+
+    # TODO find a way to link to existing job so we're not creating a new job id every time
+    job = jobs_api.create_job(json=json_info)
+    log.debug("Created job %s", job["job_id"])
+    run_id = jobs_api.run_now(
+        job_id=job["job_id"],
+        jar_params=None,
+        notebook_params=None,
+        python_params=None,
+        spark_submit_params=None,
+    )["run_id"]
+
+    runs_api = RunsApi(api_client)
+    ping_interval = int(os.getenv("AIRFLOW__ASTRO_SDK__DATABRICKS_PING_INTERVAL", "5"))
```

Comment:
This should come from the config as well, not as an environment variable

Reply:
similar to https://github.com/astronomer/astro-sdk/blob/main/python-sdk/src/astro/settings.py#L12

Reply:
Should we also have a timeout?

Reply:
@kaxil I don't think we should have a timeout. These things can take a while.